enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,670 @@
+| [Hosted Control Plane](#hosted-control-plane-change)          | Disallow HyperShift from installing on the `HighlyAvailableArbiter` and `SingleReplica` topology                |
+| [Alternative Install Flows](#alternative-install-flow-change) | Update installation flow for new node role via tooling such as Assisted Installer, Assisted Service and ZTP     |
+| [OLM Filtering](#olm-filter-addition)                         | Add support to OLM to filter operators based off of control plane topology                                      |
+
+### Infrastructure API
+
+We need an authoritative flag to make it easy for components and tools to
+identify if they are running against an `arbiter` backed HA cluster. We are
+proposing adding a new `ControlPlaneTopology` key called
+`HighlyAvailableArbiter` to denote that the cluster is installed with an
+arbiter.
+
+Currently this field will not change, once a cluster is installed as an arbiter
+HA cluster, it will remain that way. There has been discussion on making this
+field mutable to allow a cluster to transition between topologies. However, that
+conversation is beyond the scope of this enhancement, and requires its own
+dedicated enhancement. We will keep this field as static and not support
+topology migrations.
+
+### Installer Changes
+
+We will need to update the installer to have awareness of the explicit intent to
+setup the cluster with an arbiter node. Adding a new machine type similar to the
+flow for `master` and `workers` machines. As an example if the intent is to
+generate 1 `arbiter` machine and 2 `master` machines, the user specifies a
+`MachinePool` for `installConfig.arbiter` field with the desired replicas.
+
+The `arbiter` will be a `MachinePool` object that will enforce at least 1
+replica, and at least 2 replicas for the `controlPlane`. When no `arbiter` field
+is supplied regular flow will validate. When `arbiter` is supplied 2 replicas
+will be valid for for `controlPlane` as long as an `arbiter` is specified.
+
+Some validation we will enforce:
+
+1. Only on BareMetal installs.
+2. Minimum 1 Arbiter replica if Arbiter is defined.
+3. Minimum 2 Master replica when Arbiter is defined.
+
+`installConfig.yaml`
+
+```yaml
+apiVersion: v1
+baseDomain: devcluster.openshift.com
+compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+arbiter:
+  architecture: amd64
+  hyperthreading: Enabled
+  replicas: 1
+  name: arbiter
+  platform:
+    baremetal: {}
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform:
+    baremetal: {}
+  replicas: 2
+platform:
+  baremetal:
+    ...
+    hosts:
+      - name: cluster-master-0
+        role: master
+        ...
+      - name: cluster-master-1
+        role: master
+        ...
+      - name: cluster-arbiter-0
+        role: arbiter
```

Comment:
For Assisted/Agent-based installs where the role is not required, are there any heuristics we should apply for choosing the arbiter node? Presumably assisted-service should choose the smallest node (by RAM/CPU) when asked to allocate automatically?

Reply:
Yeah, that's good point, the smallest node would be the appropriate path forward, I'll capture this in the doc.